tests/test_clean.py

```diff
@@ -759,6 +759,19 @@ def test_convert_entities(data, expected):
     assert convert_entities(data) == expected
 
 
+def test_nonexistent_namespace():
+    """Verify if the namespace doesn't exist, it doesn't fail with a KeyError
+
+    The tokenizer creates "c" as a namespace and that doesn't exist in the map
+    of namespaces, so then it fails with a KeyError. I don't understand why the
+    tokenizer makes "c" into a namespace in this string.
+
+    Issue #352.
+
+    """
+    assert clean('<d {c}>') == '&lt;d c=""&gt;&lt;/d&gt;'
```

Comment:
It's interesting to note that this expected outcome suffers from the regression between Bleach 1.5 and 2.0 where the tokenizer/parser add a `</d>` and then it gets escaped. Gross.

Reply:
This is really weird to me since `d` isn't an html tag or in `ALLOWED_TAGS` and I'd expect the sanitizer to have dropped it at this point.

Reply:
Yeah, so the tokenizer or the parser (I forget which) sees an html-like tag and then closes it off. Further, it removes `{` and `}` since those don't make sense in an attributes list and it adds a value to `c`. By the time it hits the Bleach sanitizer, it looks like:

```
<d c=""></d>
```

Then the Bleach sanitizer escapes the tag since it's not in the allow list turning it into that escaped nonsense.

If we passed `strip=True`, then it'd get stripped out completely:

```
$ python
Python 3.6.3 (default, Oct  3 2017, 21:45:48) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import bleach
>>> bleach.clean('<d {c}>')
'&lt;d c=""&gt;&lt;/d&gt;'
>>> bleach.clean('<d {c}>', strip=True)
''
>>> 
```

That's what's going on. I agree it's weird. It'll get fixed when we fix the regression. I've been thinking about how to do that. I might tackle it in April along with some other things.